src/ngx_http_lua_headers.c

```diff
@@ -1240,4 +1240,17 @@ ngx_http_lua_ngx_raw_header_cleanup(void *data)
 #endif
 
 
+#if (NGX_DARWIN)
```

Comment:
This will be set for x86 macOS as well right? Then https://github.com/openresty/lua-resty-core/pull/391/files#diff-d582e95d6016d8f7a9fc6af1a80593e27b10f54e58f0441b94ca9bcd8e172d0aR57 is probably not right.

I think it will be okay if we use the new ABI for all Mac users (including x86). As @agentzh said in the comment, nobody runs OpenResty in prod using a Mac.

Reply:
Now there is no macro for `arm64` in C code, so we have to check it in Luajit.

Reply:
I think if that is the case, https://github.com/openresty/lua-resty-core/pull/391/files#diff-d582e95d6016d8f7a9fc6af1a80593e27b10f54e58f0441b94ca9bcd8e172d0aR50 should be changed to only check for macOS just for the sake of keeping the checks consistent. Otherwise we generate unused function on macOS x86 intentionally which does not feels right.

Reply:
done